src/System.Net.Http/tests/FunctionalTests/LoopbackServer.cs

```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public class LoopbackServer
+    {
+        public enum TransferType
+        {
+            None = 0,
+            ContentLength,
+            Chunked
+        }
+
+        public enum TransferError
+        {
+            None = 0,
+            ContentLengthTooLarge,
+            ChunkSizeTooLarge,
+            MissingChunkTerminator
+        }
```

Comment:
The only two that could be combined as flags would be `ChunkSizeTooLarge` and `MissingChunkTerminator`, i.e. both of the invalid conditions could be present. But I currently don't test for that. `ContentLengthTooLarge` is basically mutually exclusive with the other two chunk related errors. I didn't think it was necessary to test every combination of the chunked failures.